### PR TITLE
[Delivers #158791391] API v2 Signup Endpoint

### PR DESCRIPTION
**Pull request by JoshuaOndieki**
**What does this PR do?**
Adds signup endpoint to the API

**Description of Task to be completed**
Adding a signup endpoint to allow drivers | riders to sign up

POST `/api/v2/auth/signup`

**Tasks achieved in this PR**
- [x] POST `/api/v2/auth/signup`

**How should one test this endpoint?**
- Clone this repo using `git clone https://github.com/JoshuaOndieki/ridemyway-api.git`
- Check the ReadMe for instructions on setting up the project. 
- `git checkout ft-signup-158791391` 
- Run `nosetests ridemyway/tests/test_v2/test_auth/test_signup.py` to see the signup tests pass. Alternatively, run `pytest ridemyway/tests/test_v2/test_auth/test_signup.py`

**But why is the build failing :confused: ?**
The build is also testing endpoints which are yet to be implemented. Bear with me while I implement them.

**How can you help?**
Review my work and let me know any semantic, technical or syntax/standards errors I may have.
I will greatly appreciate any feature/change suggestions too :smiley: 

**Resources & Libs**
[This Pivotal Tracker board](https://www.pivotaltracker.com/n/projects/2179581) was used to plan and manage the tasks.

**Relevant PIvotal Tracker Stories?**
[#158791391](https://www.pivotaltracker.com/story/show/158791391)

### CODE REVIEW

**Review by LarryKarani — COMMENTED** [review]
awesome job

**Comment by meshack-mbuvi** [comment]
Why are the tests failing?

File: ridemyway/utils/db_queries.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+"""
+
+from flask import current_app as app
+
+
+def sql_signup(user):
+    sql = """INSERT INTO
+                appuser  (username, name, gender, usertype, date_joined,
+                contacts, email, password)
+                VALUES ('%s','%s','%s','%s', '%s', %d, '%s', '%s')""" % (
```
Comment (xcixor) [inline ridemyway/utils/db_queries.py]: You can also use the string.format() method to insert the data

File: ridemyway/api/v2/controllers/auth.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+
+class AuthController():
+    """
+        Controls all CRUD operations of the User object.
+    """
+
+    def signup(self, **kwargs):
+        """
+
```
Comment (kwameasiago) [inline ridemyway/api/v2/controllers/auth.py]: You have an empty comment

File: ridemyway/__init__.py [hunk 3]
```diff
@@ -47,14 +48,7 @@ def create_app(config_name):
     app.database = DATABASE
     app.config.from_object(config[config_name])
     app.config['BUNDLE_ERRORS'] = True
```
Comment (kwameasiago) [inline ridemyway/__init__.py]: Its good practice to put all your configurations in the config file

File: ridemyway/utils/db_queries.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            user.contacts,
+            user.email,
+            user.password)
+    cur = app.conn.cursor()
+    try:
+        cur.execute(sql, data)
+        app.conn.commit()
+        return True
+    except psycopg2.Error:
+        print('Roll back, user not added')
```
Comment (LWanjiru) [inline ridemyway/utils/db_queries.py]: Is this relevant to the code block?

**Review by LWanjiru — APPROVED** [review]
Minor fix, can be merged
